CONTRIBUTING.adoc

```diff
@@ -38,52 +39,49 @@ These steps should help you setup a development environment for the Globus CLI.
 your local copy of the repo. `globus` will invoke code directly out of the
 repo.
-You can write and test commands interactively in this setup.
 
-Linting & Testing
------------------
+== Running Tests
+
+We use `tox` to run our tests against all supported Python versions.
+
+You can use a tool like `pyenv` to install all of the Python versions, or
+install `uv` and the `tox-uv` plugin to have `uv` manage the interpreter versions for you.
+(`tox-uv` will also use `uv` to provision test environments.)
+
+For example:
+
+    uv tool install tox --with=tox-uv
+
+Once installed, you can run all tests with
+
+    tox run
 
-We use `tox` to run tests and `pre-commit` to run (optional) pre-commit hooks.
-Use `pre-commit install` to setup the hooks.
+[TIP]
+====
+Try `tox l` to list environments, `tox r` as a shorthand for `tox run`, and
+`tox p` to run tests in parallel.
 
-All code is autoformatted with https://github.com/ambv/black[black] and
-https://github.com/PyCQA/isort[isort]. You may run
-`make lint` to do this or configure these tools for use in your editor.
+To run a single environment, use the `-e` flag, for example: `tox r -e py3.13`.
+====
 
-Expectations for Pull Requests
-------------------------------
+== Linting, Formatting, & Type Checking
 
-  - *Make sure it merges cleanly*. We may request that you rebase if your PR
-      has merge conflicts.
-  - *List any issues closed by the pull request*
-  - *Squash intermediate and fixup commits*. We recommend running
-    `git rebase --interactive` prior to submitting a pull request.
+We use `pre-commit` to run (optional) pre-commit hooks for linting and formatting.
+Use `pre-commit install` if you want to setup the hooks locally.
 
-These are our guidelines for good commit messages:
+Code is also type-checked with `mypy`, which you can run with `tox run -e mypy`.
 
-  - No lines over 72 characters
-  - No GitHub emoji -- use your words
-  - Reference issues and pull requests where appropriate
-  - Present tense and imperative mood
+== Changelogs
 
-Additional Recommendations
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+Most changes to CLI functionality should be included in the changelog.
 
-  - All options should have descriptive help text
-  - For complex functionality, include sample usage in the helptext for the
-      relevant command. If it's a specific flag or feature that has complex
-      behavior, make it a separate section of the help text. See
-      `globus transfer async-transfer --help` for an example
-  - Comment liberally in source
-  - Every command gets its own module, named after the command name
-  - Think very hard before adding a new dependency -- keep the dependencies of
-      `globus_cli` as lightweight as possible
-  - Use the verbs `create`, `show`, `update`, and `remove` for underlying API
-      CRUD operations
+Pull Requests are required to either be labeled `no-news-is-good-news` or
+have a changelog fragment.
```

Comment:
I don't think that contributors can apply their own labels to their PRs (at a permission level) nor should they make that decision independently.

```suggestion
Pull requests must have a changelog fragment.
```

Reply:
I don't think I can square this well with `No contribution is too small. Typo and grammar fixes are welcome.`, but I see what you're getting at in that an external contributor probably needs guidance (and, at a technical level, needs assistance to apply the label).

I also don't want to lose mention of `no-news-is-good-news` since we should write down the meaning of that label somewhere.

How do you feel about... this?

> Pull Requests must have a changelog fragment or be approved by a maintainer to not need one.
>
> See  link:./changelog.d/README.adoc[the changelog doc] for information about
> how to create fragments for your changes.
>
> Maintainers use the `no-news-is-good-news` label to indicate that no changelog is needed.